### ISSUE TITLE
how to evaluate the generated factors on val set or test set?

### ISSUE THREAD

**Issue by zhengzhe97** [DulyHao/AlphaForge#7]
` data_all,data,data_valid,data_valid_withhead,data_test,data_test_withhead,_ = returned`

in train_AFF.py, line 171, train dataset(data), val dataset(data_valid) and test dateset(data_test) have been built, however, in the following code, only train dataset is used(both in train stage and eval stage). 

i am questioned why the val dataset not used in eval stage?

**Comment by DulyHao** [comment]
In our experiments, val dataset was used to select hyper-parameters. The current version code has been rearanged, in which the hyper-paramer selection part was ignored.

**Comment by DulyHao** [comment]
The whole train_AFF aims to generate a factor zoo according to the training dataset to prevent the future function, all other dataset was not used at this stage.
